src/uffd_handler/src/memory_region.rs

```diff
@@ -22,6 +34,15 @@ pub enum MemPageState {
     Anonymous,
 }
 
+/// Construct memory regions from stream received from Firecracker process.
+pub fn mem_regions_from_stream(msg: &str, mem_file_size: usize) -> Result<Vec<MemRegion>, Error> {
+    let mappings = deserialize_mappings(msg)?;
+    // Perform sanity checks on memory mappings to ensure they are not corrupted.
```

Comment:
super small nit (personal touch): looks kind of self explanatory, but it's not necessarily something that I would remove.

Also, `memory_mappings_sanity_checks` name for the `mappings_sanity_checks()` could definitely lead to the removal of that comment. (in case you choose to rename it, don't forget about the line 57, and probably lines ).

Reply:
I agree that it's self explanatory, I removed the line and renamed the function as suggested. Also, thanks for the reminder!